### PR TITLE
refact(upgrade): add control plane checks in pre-upgrade

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

This PR adds pre-upgrade checks for control plane operators.

### CODE REVIEW

File: pkg/upgrade/upgrader/cstor_cspc.go [hunk 1]
```diff
@@ -60,7 +61,11 @@ func NewCSPCPatch(opts ...CSPCPatchOptions) *CSPCPatch {
 
 // PreUpgrade ...
 func (obj *CSPCPatch) PreUpgrade() error {
-	err := obj.CSPC.PreChecks(obj.From, obj.To)
+	err := isOperatorUpgraded("cspc-operator", obj.Namespace, obj.To, obj.KubeClientset)
```
Comment (mittachaitu) [inline pkg/upgrade/upgrader/cstor_cspc.go:64]: Should it go into precheck right(because it is mostly like check before upgrade)?

Comment (shubham14bajpai) [reply pkg/upgrade/upgrader/cstor_cspc.go:64]: Precheck is for a specific CR, it does not checks for any other resource. Preupgrade is for a volume or pool as a whole(means openebs CRs + deploy + service), so it makes more sense to have it in preupgrade.

File: pkg/upgrade/upgrader/helper.go [hunk 2]
```diff
@@ -62,3 +64,25 @@ func GetPatchData(oldObj, newObj interface{}) ([]byte, error) {
...
+			LabelSelector: "openebs.io/component-name=" + componentName,
+		})
+	if err != nil {
+		return err
+	}
+	if len(operatorPods.Items) == 0 {
+		return fmt.Errorf("operator pod missing for %s", componentName)
+	}
+	for _, pod := range operatorPods.Items {
+		if pod.Labels["openebs.io/version"] != toVersion {
```
Comment (mittachaitu) [inline pkg/upgrade/upgrader/helper.go:82]: nit: We can move `openebs.io/version` to some global constant.

Comment (shubham14bajpai) [reply pkg/upgrade/upgrader/helper.go:82]: Sure. Will take this in separate PR.

**Pull request merged**